lib/board.rb

```diff
@@ -0,0 +1,70 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+    #binding.pry
+  end
+
+  def reset!
+    @cells = Array.new(9, " ") #[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    #binding.pry
+  end
+
+  def display
+    counter = 0
+    @cells.each do |space|
+      counter += 1
+      if counter == 3 || counter == 6 || counter == 9
+        print " #{space} "
+      else
+        print " #{space} |"
+      end
+      if counter % 3 == 0
+        if counter == 9
+          break
+        else
+          print "\n"
+          puts "-----------"
+        end
+      end
+    end
+  end
+
+  def position(input)
+    input = input.to_i - 1
+    #binding.pry
+    return "#{@cells[input]}"
+  end
+
+  def full?
+    @cells.detect {|space| space == " "} ? false : true
+  end
+
+  def turn_count
+    turn_count = Array.new
+    turn_count = @cells.find_all {|space| space == "X" || space == "O"}
+    return turn_count.size
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```